protocol/testing/e2e/gov/vest_test.go

```diff
@@ -106,149 +288,212 @@ func TestSetVestEntry(t *testing.T) {
 
 	for name, tc := range tests {
 		t.Run(name, func(t *testing.T) {
-			tApp := testapp.NewTestAppBuilder().WithGenesisDocFn(func() (genesis types.GenesisDoc) {
+			tApp := testapp.NewTestAppBuilder(t).WithGenesisDocFn(func() (genesis types.GenesisDoc) {
 				genesis = testapp.DefaultGenesis()
 				testapp.UpdateGenesisDocWithAppStateForModule(
 					&genesis,
 					func(genesisState *govtypesv1.GenesisState) {
 						genesisState.Params.VotingPeriod = &testapp.TestVotingPeriod
 					},
 				)
-				// If updating an existing vest entry, initialize vest module with a vest entry that has the
-				// same key. Otherwise, initialize vest module with no vest entry.
-				genesisVestEntries := []vesttypes.VestEntry{}
-				if tc.updateExistingVestEntry {
-					genesisVestEntries = append(genesisVestEntries, vesttypes.VestEntry{
-						VesterAccount:   tc.msg.Entry.VesterAccount,
-						TreasuryAccount: tc.msg.Entry.TreasuryAccount + "_initial",
-						Denom:           tc.msg.Entry.Denom + "_initial",
-						StartTime:       tc.msg.Entry.StartTime.AddDate(0, 0, 1),
-						EndTime:         tc.msg.Entry.EndTime.AddDate(0, 0, 1),
-					})
-				}
+				return genesis
+			}).Build()
+			ctx := tApp.InitChain()
+			initialVestEntries := tApp.App.VestKeeper.GetAllVestEntries(ctx)
+
+			// Submit and tally governance proposal that includes `MsgSetVestEntry`s.
+			ctx = testapp.SubmitAndTallyProposal(
+				t,
+				ctx,
+				tApp,
+				tc.msgs,
+				tc.expectCheckTxFails,
+				tc.expectSubmitProposalFail,
+				govtypesv1.ProposalStatus_PROPOSAL_STATUS_FAILED,
+			)
+
+			// Verify that vest entries in state match the ones before proposal submission.
+			require.Equal(t, initialVestEntries, tApp.App.VestKeeper.GetAllVestEntries(ctx))
+		})
+	}
+}
+
+func TestDeleteVestEntry_Success(t *testing.T) {
+	tests := map[string]struct {
+		msgs                 []sdk.Msg
+		genesisVestEntryKeys []string // keys of vest entries in genesis state.
+	}{
+		"Success: delete one vest entry": {
+			msgs: []sdk.Msg{
+				&vesttypes.MsgDeleteVestEntry{
+					Authority:     authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					VesterAccount: "random_vester",
```

Comment:
Nit: Can we define "random_vester" as a constant string (e.g. "TEST_VESTERT_ACCOUNT1 = "random_vester")